### PR TITLE
Submission 2

### PR DESCRIPTION
**Pull request by dimascatur**
This project is submitted to complete the submission task, beg for the review.

### CODE REVIEW

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/league/detail/page/SectionsPagerAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import androidx.fragment.app.FragmentPagerAdapter
+
+/**
+ * A [FragmentPagerAdapter] that returns a fragment corresponding to
+ * one of the sections/tabs/pages.
+ */
+class SectionsPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {
+
+    val fragmentList = arrayListOf<Fragment>()
+    val fragmentTitle = arrayListOf<String>()
```
Comment (KhrisnaIndrawan) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/league/detail/page/SectionsPagerAdapter.kt]: Functions or variables that are only used in the same class should be made private.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import com.dicoding.picodiploma.sejiwaproject.features.match.detail.model.LogoTeam
+import com.dicoding.picodiploma.sejiwaproject.commons.utils.invisible
+import com.dicoding.picodiploma.sejiwaproject.commons.utils.visible
+import com.google.gson.Gson
+import kotlinx.android.synthetic.main.activity_detail_match.*
+
+class DetailMatchActivity : AppCompatActivity(),
+    DetailMatchView {
+
+    private var matchs: MutableList<DetailMatch> = mutableListOf()
```
Comment (KhrisnaIndrawan) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: Typo, change variable name to "matches". Use shift+f6 to refactor variable/function name.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    private lateinit var presenter: NextMatchPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        pageViewModel = ViewModelProviders.of(this).get(PageViewModel::class.java).apply {
+            setIndex(arguments?.getInt(ARG_SECTION_NUMBER) ?: 2)
+        }
+    }
+
+    override fun onCreateView(
```
Comment (KhrisnaIndrawan) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt]: The onCreateView function should only be used to inflate view. For initializing the view, it should be done inside the onViewCreated function.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            .appendPath("v1")
+            .appendPath("json")
+            .appendPath(BuildConfig.TSDB_API_KEY)
+            .appendPath("eventsnextleague.php")
+            .appendQueryParameter("id", idEvent)
+            .build()
+            .toString()
+    }
+
+    fun getMatchDetail(idEvent: String?): String {
```
Comment (alfianyusufabdullah) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt]: Don't forget to reformat code.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import com.google.gson.Gson
+import kotlinx.android.synthetic.main.fragment_next_match.*
+
+/**
+ * A simple [Fragment] subclass.
+ */
+class NextMatchFragment : Fragment(),
+    NextMatchView {
+    private lateinit var rvMatch: RecyclerView
+    private lateinit var pageViewModel: PageViewModel
```
Comment (alfianyusufabdullah) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt]: This variable is not really needed, should be to remove.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/PreviousMatchFragment.kt [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+import com.google.gson.Gson
+import kotlinx.android.synthetic.main.fragment_next_match.*
+
+/**
+ * A placeholder fragment containing a simple view.
+ */
+class PreviousMatchFragment : Fragment(),
+    PreviousMatchView {
+    private lateinit var rvMatch: RecyclerView
+    private lateinit var pageViewModel: PageViewModel
```
Comment (alfianyusufabdullah) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/PreviousMatchFragment.kt]: This variable is not really needed, should be to remove.

File: app/build.gradle [hunk 7]
```diff
@@ -42,4 +48,7 @@ dependencies {
     implementation "org.jetbrains.anko:anko-recyclerview-v7:$anko_version"
     implementation 'de.hdodenhof:circleimageview:3.0.1'
     implementation 'com.github.bumptech.glide:glide:4.10.0'
+
+    implementation 'com.squareup.picasso:picasso:2.71828'
```
Comment (alfianyusufabdullah) [inline app/build.gradle]: This library is not use, you can remove it

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/league/detail/page/SectionsPagerAdapter.kt (hunk 1 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/league/detail/page/SectionsPagerAdapter.kt]: resolved, thanks

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt (hunk 2 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: resolved, thanks

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt (hunk 4 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt]: resolved, thanks

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/PreviousMatchFragment.kt (hunk 6 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/PreviousMatchFragment.kt]: resolved, thanks

File: app/build.gradle (hunk 7 above)
Comment (dimascatur) [reply app/build.gradle]: resolved, thanks

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt (hunk 3 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt]: resolveld, thanks

File: app/build.gradle [hunk 8]
```diff
@@ -42,4 +48,6 @@ dependencies {
     implementation "org.jetbrains.anko:anko-recyclerview-v7:$anko_version"
     implementation 'de.hdodenhof:circleimageview:3.0.1'
     implementation 'com.github.bumptech.glide:glide:4.10.0'
+
```
Comment (nrohmen) [inline app/build.gradle]: over spacing

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt [hunk 9]
```diff
@@ -0,0 +1,79 @@
+package com.dicoding.picodiploma.sejiwaproject.commons.api
+
+import android.net.Uri
+import com.dicoding.picodiploma.sejiwaproject.BuildConfig
+
+object TheSportDBApi {
+    fun getDetails(idleague: String?): String {
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt]: Use camelCase. `idLeague`

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+        return Uri.parse(BuildConfig.BASE_URL).buildUpon()
+            .appendPath("api")
+            .appendPath("v1")
+            .appendPath("json")
+            .appendPath(BuildConfig.TSDB_API_KEY)
+            .appendPath("searchevents.php")
+            .appendQueryParameter("e", event)
+            .build()
+            .toString()
+
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/commons/api/TheSportDBApi.kt]: Remove this space.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+        val gson = Gson()
+        presenter =
+            DetailMatchPresenter(
+                this,
+                request,
+                gson
+            )
+        presenter.getDetailMatch(matchDetail)
+
+
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/detail/DetailMatchActivity.kt]: Over spacing

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchAdapter.kt [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+            .load(match.badgeAway)
+            .into(holder.awayLogo)
+
+        holder.itemView.setOnClickListener {
+            it.context.startActivity<DetailMatchActivity>(EXTRA_ID to match.matchId)
+
+        }
+    }
+
+
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchAdapter.kt]: Over spacing.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+
+/**
+ * A simple [Fragment] subclass.
+ */
+class NextMatchFragment : Fragment(),
+    NextMatchView {
+    private lateinit var rvMatch: RecyclerView
+    private lateinit var presenter: NextMatchPresenter
+
+
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt]: Over spacing.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+        val gson = Gson()
+        presenter =
+            NextMatchPresenter(
+                this,
+                request,
+                gson
+            )
+        presenter.getNextMatch(id ?: "4328")
+
+
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt]: Over spacing.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+                .apply {
+                arguments = Bundle().apply {
+                    putString(ID_LEAGUE, idLeague)
+                }
+            }
+        }
+    }
+
+
+
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/next/NextMatchFragment.kt]: Over spacing.

File: app/src/main/res/layout/item_leagues.xml [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+        android:layout_height="100dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginRight="16dp"
+        tools:src="@drawable/english" />
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_centerVertical="true"
+        android:layout_toEndOf="@id/league_logo"
+        android:layout_toRightOf="@id/league_logo"
```
Comment (nrohmen) [inline app/src/main/res/layout/item_leagues.xml]: You already defining `EndOf` attribute, remove this line.

File: app/src/main/res/layout/item_leagues.xml [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+    android:layout_height="wrap_content"
+    android:orientation="horizontal"
+    android:padding="16dp">
+
+    <ImageView
+        android:id="@+id/league_logo"
+        android:layout_width="100dp"
+        android:layout_height="100dp"
+        android:layout_marginEnd="16dp"
+        android:layout_marginRight="16dp"
```
Comment (nrohmen) [inline app/src/main/res/layout/item_leagues.xml]: You already defining `merginEnd` attribute, remove this line.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/search/SearchMatchActivity.kt [hunk 18]
```diff
@@ -0,0 +1,86 @@
+package com.dicoding.picodiploma.sejiwaproject.features.match.search
+
+import android.app.SearchManager
+import android.content.Context
+import android.os.Bundle
+import android.view.Menu
+import android.widget.Toast
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/search/SearchMatchActivity.kt]: Unused import.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/model/Matchs.kt [hunk 19]
```diff
@@ -0,0 +1,39 @@
+package com.dicoding.picodiploma.sejiwaproject.features.match.previous.model
+
+import android.os.Parcelable
+import com.google.gson.annotations.SerializedName
+import kotlinx.android.parcel.Parcelize
+
+@Parcelize
+data class Matchs(
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/model/Matchs.kt]: Matches.

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/model/MatchsResponse.kt [hunk 20]
```diff
@@ -0,0 +1,5 @@
+package com.dicoding.picodiploma.sejiwaproject.features.match.previous.model
+
+data class MatchsResponse(
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/model/MatchsResponse.kt]: `MatchesResponse`

Comment (nrohmen) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/model/MatchsResponse.kt]: or MatchResponses

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/PreviousMatchFragment.kt [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+    override fun showLoading() {
+        progress_bar.visible()
+    }
+
+    override fun hideLoading() {
+        progress_bar.invisible()
+    }
+
+    override fun showPreviousMatch(data: List<Matchs>) {
+        Log.d("size", "" + data.size)
```
Comment (nrohmen) [inline app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/PreviousMatchFragment.kt]: Delete the log after it's finished.

File: app/src/main/res/layout/item_leagues.xml (hunk 17 above)
Comment (dimascatur) [reply app/src/main/res/layout/item_leagues.xml]: resolved, thanks

File: app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/model/Matchs.kt (hunk 19 above)
Comment (dimascatur) [reply app/src/main/java/com/dicoding/picodiploma/sejiwaproject/features/match/previous/model/Matchs.kt]: resolved, thanks

**Pull request merged**